app/assets/javascripts/components/components/status_content.jsx

```diff
@@ -93,7 +93,7 @@ const StatusContent = React.createClass({
     const { status } = this.props;
     const { hidden } = this.state;
 
-    const content = { __html: emojify(status.get('content')) };
+    const content = { __html: emojify(status.get('content').replace(/\s*(<p>|<\/p>)\s*/g, '$1').replace(/\s{2}/g, ' &nbsp;')) };
```

Comment:
It's replacing two spaces with one regular space and one `&nbsp;` so the result is the same but without relying on `pre-wrap`

The regex does not strip out `p` tags, only whitespace that gets inserted before and after them in current versions of mastodon which can cause unintended indentation.